src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java

```diff
@@ -22,23 +25,46 @@
     private static final Logger LOGGER = LoggerFactory
         .getLogger(CompanyAccountsApplication.APPLICATION_NAME_SPACE);
 
+    private static final String FILING_CONTROLLER_ERROR = "FilingController error:";
+
     @Autowired
     private FilingService filingService;
 
     @GetMapping("/private/transactions/{transactionId}/company-accounts/{companyAccountId}/filings")
     public ResponseEntity generateFiling(@PathParam("transactionId") String transactionId,
         @PathParam("companyAccountId") String accountId, HttpServletRequest request) {
 
+        Transaction transaction =
+            (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        if (transaction == null) {
```

Comment:
Isn't this stuff done in the interceptor?

Reply:
it should be done in the controller but the changes are not there yet. For that reason, I've added this check here.

Reply:
It is. I will remove it.